optimus/dataframe/columns.py

```diff
@@ -0,0 +1,1154 @@
+import unicodedata
+import string
+import re
+from functools import reduce
+import builtins
+
+from multipledispatch import dispatch
+
+from pyspark.ml.feature import Imputer
+from pyspark.ml.feature import QuantileDiscretizer
+from pyspark.sql import DataFrame
+from pyspark.sql import functions as F
+from pyspark.sql.functions import Column
+
+from pyspark.ml.linalg import Vectors, VectorUDT
+from pyspark.ml.feature import VectorAssembler
+
+# Helpers
+from optimus.helpers.constants import *
+from optimus.helpers.decorators import add_attr, add_method
+from optimus.helpers.checkit \
+    import is_num_or_str, is_list, is_, is_tuple, is_list_of_dataframes, is_list_of_tuples, \
+    is_function, is_one_element, is_type, is_int
+
+from optimus.helpers.functions \
+    import validate_columns_names, parse_columns, parse_spark_dtypes, collect_to_dict, format_dict, \
+    tuple_to_dict, val_to_list, filter_list
+
+from optimus.functions import filter_row_by_data_type as fbdt
+from optimus.functions import abstract_udf as audf, concat
+
+from optimus.helpers.raiseit import RaiseIfNot
+
+
+def cols(self):
+    @add_attr(cols)
+    @dispatch(str, object)
+    def append(col_name=None, value=None):
+        """
+        Append a column to a Dataframe
+        :param col_name:
+        :param value:
+        :return:
+        """
+
+        def lit_array(value):
+            temp = []
+            for v in value:
+                temp.append(F.lit(v))
+            return F.array(temp)
+
+        df = self
+
+        if is_num_or_str(value):
+            value = F.lit(value)
+        elif is_list(value):
+            value = lit_array(value)
+        elif is_tuple(value):
+            value = lit_array(list(value))
+
+        if is_(value, Column):
+            df = df.withColumn(col_name, value)
+
+        return df
+
+    @add_attr(cols)
+    @dispatch(list)
+    def append(cols_values=None):
+        """
+        Append a column or a Dataframe to a Dataframe
+        :param cols_values:
+        :return:
+        """
+
+        # Append a dataframe
+        if is_list_of_dataframes(cols_values):
+            dfs = cols_values
+            dfs.insert(0, self)
+            df_result = concat(dfs, like="columns")
+
+        elif is_list_of_tuples(cols_values):
+            df_result = self
+            for c in cols_values:
+                col_name = c[0]
+                value = c[1]
+                df_result = df_result.cols.append(col_name, value)
+
+        return df_result
```

Comment:
Here what happens if is not a list or a tuple? PyCharm is telling me `Local variable 'df_result' might be referenced before assignment`